util/uploader.ts

```diff
@@ -0,0 +1,323 @@
+import { UUID } from 'angular2-uuid';
+import { MimeTypeService } from './mime-type.service';
+
+const KB = 1024;
+const MB = 1024 * KB;
+const GB = 1024 * MB;
+
+// Status of `FileUpload`s and `FilePart`s
+const enum UploadStatus {
+  waiting,
+  uploading,
+  complete,
+  failed
+}
+
+// This is per ENV, put into a single class for convenience, typing, and clarity
+export class UploaderConfiguration {
+  constructor(
+    public bucketName:string,
+    public signUrl:string,
+    public awsKey:string,
+    public awsUrl:string = 'https://s3.amazonaws.com',
+    public useCloudfront:boolean = false
+  ) {}
+}
+
+export class Uploader {
```

Comment:
that is not being used at the moment - that is my own ts version of what evaporate does that I was about 75% complete on when I finally got evaporate working. I checked it in here to keep track of if/when I do go back to it.